src/jit/lowerxarch.cpp

```diff
@@ -3869,27 +3897,103 @@ bool Lowering:: IsContainableImmed(GenTree* parentNode, GenTree* childNode)
     return true;
 }
 
-//----------------------------------------------------------------------
-// TryToSetRegOptional - sets a bit to indicate to LSRA that register
-// for a given tree node is optional for codegen purpose.  If no
-// register is allocated to such a tree node, its parent node treats
-// it as a contained memory operand during codegen.
+//-----------------------------------------------------------------------
+// IsProfitableToMarkOp1AsRegOptional: returns true if it is likely to 
```

Comment:
Suggestion: This name doesn't make it really clear that if it returns false we should use op2, An alternative would be to call it something like "PreferredRegOptionalOperand", and return either op1 or op2. Then, the callees can either call `SetRegOptional` on the return value, or simply check if it is equal to op1.